server/server.go

```diff
@@ -88,16 +83,23 @@ func NewServer(addr string, opts ...Option) *Server {
 
 	server.handler = &http.Server{
 		Addr:              addr,
-		ReadHeaderTimeout: ReadHeaderTimeout,
-		ReadTimeout:       ReadTimeout,
+		ReadHeaderTimeout: serverConfig.ReadHeaderTimeout,
+		ReadTimeout:       serverConfig.ReadTimeout,
```

Comment:
How default values will be assigned?

Reply:
we will have to check if base ctx contains `Config` key in its values, if not apply the default config